beacon-chain/sync/initial-sync/service.go

```diff
@@ -420,50 +405,42 @@ func (s *InitialSync) processState(msg p2p.Message) {
 	ctx, span := trace.StartSpan(msg.Ctx, "beacon-chain.sync.initial-sync.processState")
 	defer span.End()
 	data := msg.Data.(*pb.BeaconStateResponse)
-	beaconState := data.BeaconState
+	finalizedState := data.FinalizedState
+	justifiedState := data.JustifiedState
+	canonicalState := data.CanonicalState
 	recState.Inc()
 
-	if s.currentSlot > beaconState.FinalizedEpoch*params.BeaconConfig().SlotsPerEpoch {
+	if s.currentSlot > finalizedState.FinalizedEpoch*params.BeaconConfig().SlotsPerEpoch {
```

Comment:
shouldn't this be `finalizedState.CurrentEpoch`, it would make sense if it's `canonicalState.FinalizedEpoch`, the finalized state is already finalized